File: 66-PlusOne/66-PlusOne.cpp
// Last updated: 6/2/2025, 6:41:13 PM
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        for (int i = digits.size() - 1; i >= 0; --i) {
            ++digits[i];
            digits[i] %= 10;
            if (digits[i] != 0) return digits;
        }
        digits.insert(digits.begin(), 1);
        return digits;
    }
};

File: 66-PlusOne/66-PlusOne.py
# Last updated: 1/1/2026, 8:58:39 AM
1class Solution:
2  def plusOne(self, digits: list[int]) -> list[int]:
3    for i, d in reversed(list(enumerate(digits))):
4      if d < 9:
5        digits[i] += 1
6        return digits
7      digits[i] = 0
8
9    return [1] + digits